test/secret_store.go

```diff
@@ -0,0 +1,84 @@
+package test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/pborman/uuid"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/libopenstorage/secrets"
+)
+
+type storeTest struct {
+	s                   secrets.SecretStore
+	secretIdWithData    string
+	secretIdWithoutData string
+}
+
+func RunForStore(store secrets.SecretStore, t *testing.T) {
+	st := &storeTest{
+		s:                   store,
+		secretIdWithData:    "openstorage_secret_data_" + uuid.New(),
+		secretIdWithoutData: "openstorage_secret_" + uuid.New(),
+	}
+
+	st.TestPutSecret(t)
+	st.TestGetSecret(t)
+	st.TestDeleteSecret(t)
+}
+
+func (a *storeTest) TestPutSecret(t *testing.T) {
+	secretData := make(map[string]interface{})
+	secretData["key1"] = "value1"
+	secretData["key2"] = "value2"
+	// PutSecret with non-nil secretData
+	err := a.s.Set(context.Background(), secrets.SecretKey{Name: a.secretIdWithData}, secretData)
+	assert.NoError(t, err, "Unexpected error on PutSecret")
+
+	// PutSecret with nil secretData
+	err = a.s.Set(context.Background(), secrets.SecretKey{Name: a.secretIdWithoutData}, nil)
+	assert.NoError(t, err, "Expected PutSecret to succeed. Failed with error: %v", err)
+}
+
+func (a *storeTest) TestGetSecret(t *testing.T) {
+	// GetSecret with non-existant id
```

Comment:
nit: you should add a test where SecretKey with a Prefix is set and check that such a Get API fails. And also add a Put equivalent when Prefix is set.